### ISSUE TITLE
decoder/train.py: extremely large latent_loss when training Qwen2.5/Qwen3.5 aligner, is this expected?

### ISSUE THREAD

**Issue by jingfengou** [hwanyu112/Latent-Sketchpad#4]
Hi, thanks for open-sourcing this project.
I am adapting the decoder/aligner training pipeline for Qwen-based vision backbones, and I want to confirm whether very large `latent_loss` values are expected in `decoder/train.py`.
## Context
I am training the dense aligner (`ClipToLatentAligner`) in:
- `decoder/train.py`
- `decoder/aligner/dense_aligner.py`
The training objective is:
```python
loss = reconstruction_loss + latent_loss + embed_loss
```
where:
- image_loss / reconstruction_loss comes from decoded image reconstruction
- embed_loss is MSE between aligner embeddings and VAE embeddings
- latent_loss is computed with gaussian_nll_loss(...)
What I observe
For Qwen-based vision encoders, the total loss is dominated by latent_loss, which becomes extremely large.
Example from a short Qwen2.5 smoke run:
- step=5
  - train_loss=361100032.0000
  - latent_loss=361099968.0000
  - embed_loss=0.9682
  - image_loss=48.1484
- step=10
  - train_loss=225297568.0000
  - latent_loss=225297520.0000
  - embed_loss=1.1181
  - image_loss=51.8171
Final summary:
- train_loss: 134009968.000
- latent_loss: 134009896.000
- embed_loss: 1.040
- image_loss: 73.881
So:
- embed_loss looks normal
- image_loss is not tiny but still on a reasonable scale
- latent_loss is huge and completely dominates total loss
What I already checked
1. Vision tower loading
Initially I found that my Qwen2.5 visual encoder loading path was wrong, which caused many MISSING/UNEXPECTED weights.
I fixed that by loading the full model first and then taking the visual tower, similar to Qwen3.5:
- for Qwen2.5: load Qwen2_5_VLForConditionalGeneration, then use model.model.visual
- for Qwen3.5: load Qwen3_5ForConditionalGeneration, then use model.model.visual
After this fix, the Qwen2.5 load report no longer shows missing/unexpected vision weights.
So the very large latent_loss does not seem to come from a broken vision checkpoint load.
2. This is reproducible even in a small smoke run
I reproduced this on a very small setup, so it does not seem to be just a long-run instability issue.
### 3. The problem seems specific to `latent_loss`
`embed_loss` and `image_loss` are much smaller, while `latent_loss` is on the order of `1e8`.
This makes me wonder whether:
- this scale is actually expected for the current `gaussian_nll_loss` setup, or
- something is numerically off in the latent distribution alignment part
## Relevant code
In `decoder/train.py`:
```python
latent_loss = torch.clamp(
    F.gaussian_nll_loss(
        latent_output.latent_dist.mean,
        target_latents.sample().to(self.device),
        torch.clamp(latent_output.latent_dist.std**2, min=1e-6).to(self.device),
        eps=1e-5,
        reduction='mean'
    ),
    0
)
if self.need_dense:
    latent_loss += F.mse_loss(
        latent_output.latent_dist.std,
        target_latents.std.to(self.device)
    )
```
And in decoder/aligner/dense_aligner.py, the aligner predicts a latent distribution through:
- transformer encoder/decoder blocks
- conv_out
- quant_conv
- DiagonalGaussianDistribution(output)

##
My questions
1. Is it expected that latent_loss is extremely large (e.g. 1e8) in this training pipeline?
2. Did you observe similar loss scales when training the official decoder/aligner?
3. Should I compare trends only, rather than absolute magnitude, for latent_loss?
4. Is gaussian_nll_loss(..., reduction="mean") here known to produce very large values for some vision backbones?
5. Would you recommend any normalization, rescaling, or reformulation for this loss when adapting to Qwen-based visual features?
Additional context
I am using:
- SDXL VAE as in the provided training code
- dense aligner
- Qwen-based visual features
- image size 224
- aligner hidden dim 512
- 12 layers
Thanks!
